### ISSUE TITLE
[Question]: Greyscale preprocessor?

### ISSUE THREAD

**Issue by SLAPaper** [Mikubill/sd-webui-controlnet#1427]
Is there a preprocessor that can convert input image to greyscale? Then no need to do it externally in other picture editing tools.

To use with the new third-party greyscale colorization model: [Brightness Control](https://huggingface.co/ioclab/ioc-controlnet/resolve/main/models/control_v1p_sd15_brightness.safetensors)

More info of the model: https://huggingface.co/ioclab/control_v1p_sd15_brightness

**Comment by SLAPaper** [comment]
Demo images of the model, from document(Chinese): https://aigc.ioclab.com/sd-showcase/brightness-controlnet.html

![scene](https://aigc.ioclab.com/_images/13.jpg)
![old photo](https://aigc.ioclab.com/_images/21.jpg)
![anime](https://aigc.ioclab.com/_images/31.jpg)
![3D model](https://aigc.ioclab.com/_images/41.jpg)

**Comment by dareima** [comment]
Anyone knows if there's a .yaml file? A1111 will always throw this error when loading the brightness model.

ERROR: ControlNet cannot find model config [D:\Programme\A1111\stable-diffusion-webui\extensions\sd-webui-controlnet\models\control_v1p_sd15_brightness.yaml]

Can't find a .yaml file anywhere on the net though. The model still works but I'm wondering if the results would be better with the proper .yaml file.

**Comment by SLAPaper** [comment]
> Anyone knows if there's a .yaml file? A1111 will always throw this error when loading the brightness model.
> 
> ERROR: ControlNet cannot find model config [D:\Programme\A1111\stable-diffusion-webui\extensions\sd-webui-controlnet\models\control_v1p_sd15_brightness.yaml]
> 
> Can't find a .yaml file anywhere on the net though. The model still works but I'm wondering if the results would be better with the proper .yaml file.

I simply copy the file from https://github.com/lllyasviel/ControlNet/tree/main/models and rename it, then it has no complaint about that.

**Comment by SLAPaper** [comment]
> 

then maybe you should ask ioclab directly?

**Comment by Razunter** [comment]
> I simply copy the file from https://github.com/lllyasviel/ControlNet/tree/main/models and rename it, then it has no complaint about that.

I'm getting error with the default one:
```
RuntimeError: Error(s) in loading state_dict for ControlNet:
        Missing key(s) in state_dict: "time_embed.0.weight", "time_embed.0.bias", "time_embed.2.weight",
        ```
etc...

**Comment by Subarasheese** [comment]
@SLAPaper @dareima Did you manage to make it work with A1111? Which preprocessor / settings did you use?
I also set the yaml as suggested, but like @Razunter I am facing the same problem.

**Comment by dareima** [comment]
> @SLAPaper @dareima Did you manage to make it work with A1111? Which preprocessor / settings did you use? I also set the yaml as suggested, but like @Razunter I am facing the same problem.

For now, I simply made a copy of my 'stable-diffusion-webui\extensions\sd-webui-controlnet\models\cldm_v15.yaml' and renamed it to 'control_v1p_sd15_brightness.yaml'. This eliminates the error for me. As there is still no specific .yaml around for the brightness model, I suppose none is required.

When using the brightness model I'm not using a preprocessor (none). To give an example for using it to colorize monochrome pictures, I have attached two pngs. Load them into PNG info of A1111 to see the settings.

![00005-3730067823](https://github.com/Mikubill/sd-webui-controlnet/assets/64737959/35b40dd9-017f-49a1-8ccc-8e8365d41e41)
![00008-1430860838](https://github.com/Mikubill/sd-webui-controlnet/assets/64737959/db839756-872d-4600-8c7b-90dc62428073)
